src/test/java/org/prebid/server/bidder/gothamads/GothamAdsBidderTest.java

```diff
@@ -0,0 +1,246 @@
+        final BidRequest bidRequest = givenBidRequest(impBuilder -> impBuilder.ext(invalidExt));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = sut.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1).first()
+                .satisfies(error -> assertThat(error.getType()).isEqualTo(Type.bad_input));
+    }
+
+    @Test
+    public void shouldMakeCorrectRequest() {
+        // given
+        final String secondImpId = "imp_id2";
+        final BidRequest bidRequest = givenBidRequest(identity(), i -> i.id(secondImpId).ext(givenExt()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = sut.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        final BidRequest expectedBidRequest = givenBidRequest(i -> i.ext(null), i -> i.id(secondImpId).ext(givenExt()));
+        assertThat(result.getValue()).usingRecursiveComparison()
+                .isEqualTo(expectedResults(expectedBidRequest, IMP_ID, secondImpId).getValue());
+    }
+
+    @Test
+    public void shouldReturnErrorWhileMakingBidsWhenResponseCanNotBeParsed() {
+        // given
+        final BidderCall<BidRequest> httpCall = givenHttpCall(null, "invalid");
+
+        // when
+        final Result<List<BidderBid>> actual = sut.makeBids(httpCall, null);
+
+        // then
+        assertThat(actual.getValue()).isEmpty();
+        assertThat(actual.getErrors()).containsExactly(badServerResponse("Bad Server Response"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("responsesWithoutSeatBids")
+    public void shouldReturnErrorWhileMakingBidsWhenResponseDoesNotHaveSeatBid(String response) {
+        // given
+        final BidderCall<BidRequest> httpCall = givenHttpCall(null, response);
+
+        // when
+        final Result<List<BidderBid>> actual = sut.makeBids(httpCall, null);
+
+        // then
+        assertThat(actual.getValue()).isEmpty();
+        assertThat(actual.getErrors()).containsExactly(badServerResponse("Empty SeatBid array"));
+    }
+
+    @Test
+    public void shouldReturnBids() {
+        // given
+        final Bid bannerBid = Bid.builder().impid("1").mtype(1).build();
+        final Bid videoBid = Bid.builder().impid("2").mtype(2).build();
+        final Bid nativeBid = Bid.builder().impid("3").mtype(4).build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                givenBidRequest(identity()),
+                givenBidResponse(bannerBid, videoBid, nativeBid)
+        );
+
+        // when
+        final Result<List<BidderBid>> result = sut.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).containsExactly(
+                BidderBid.of(bannerBid, banner, null),
+                BidderBid.of(videoBid, video, null),
+                BidderBid.of(nativeBid, xNative, null)
+        );
```

Comment:
Please, create separate test for each `mtype`. Unit tests should be as granular as possible.

Reply:
I believe this test is granular enough itself, except you suggest adding more tests cases for bids